renku/core/metadata/database.py

```diff
@@ -591,12 +611,13 @@ def _serialize_helper(self, object):
         elif isinstance(object, list):
             return [self._serialize_helper(value) for value in object]
         elif isinstance(object, dict):
+            result = dict()
             for key, value in object.items():
-                object[key] = self._serialize_helper(value)
-            return object
+                result[key] = self._serialize_helper(value)
+            return result
         elif isinstance(object, Index):
             # NOTE: Index objects are not stored as references and are included in their parent object (i.e. root)
-            state = object.__getstate__()
+            state = object.__getstate__().copy()
```

Comment:
I had bugs with this where it leaked the internal state of object, and then updating `state` actually modified the `object`

Reply:
On a second look, I think I just added it here because I had to add it in other place and this looked like it would have the same issue, but you're right